### PR TITLE
new package: py-zarr

### PR DESCRIPTION
**Pull request by Sinan81**
The missing dependencies will be added shortly.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-zarr/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    version('2.3.2', sha256='c62d0158fb287151c978904935a177b3d2d318dea3057cfbeac8541915dfa105')
+
+    depends_on('python@3.5:',               type=('build', 'run'))
+    depends_on('py-asciitree',              type=('build', 'run'))
+    depends_on('py-fasteners',              type=('build', 'run'))
+    depends_on('py-msgpack',                type=('build', 'run'))
+    depends_on('py-setuptools@38.6.0:',     type='build')
+    depends_on('py-setuptools-scm@1.5.4:',  type='build')
+    depends_on('py-numcodecs@0.6.4:',       type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zarr/package.py]:
```suggestion
    depends_on('py-numcodecs@0.6.2:',       type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-zarr/package.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class PyZarr(PythonPackage):
+    """Zarr is a Python package providing an implementation of chunked,
+    compressed, N-dimensional arrays."""
+
+    homepage = "https://zarr.readthedocs.io"
+    url      = "https://pypi.io/packages/source/z/zarr/zarr-2.3.2.tar.gz"
+
+    version('2.3.2', sha256='c62d0158fb287151c978904935a177b3d2d318dea3057cfbeac8541915dfa105')
+
+    depends_on('python@3.5:',               type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zarr/package.py]: Drop this line, older versions of Python are indeed supported and require additional dependencies you'll have to add.

File: var/spack/repos/builtin/packages/py-zarr/package.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on('py-asciitree',              type=('build', 'run'))
+    depends_on('py-fasteners',              type=('build', 'run'))
+    depends_on('py-msgpack',                type=('build', 'run'))
+    depends_on('py-setuptools@38.6.0:',     type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zarr/package.py]:
```suggestion
    depends_on('py-setuptools@18.1:',     type='build')
```

File: var/spack/repos/builtin/packages/py-zarr/package.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on('py-setuptools-scm@1.5.4:',  type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zarr/package.py]:
```suggestion
    depends_on('py-setuptools-scm@1.5.5:',  type='build')
```

**Comment by Sinan81** [comment]
Thanks Adam! It turns out I got requirements from setup.py on master branch of zarr as opposed to the version indicated in the Spack package file!

**Comment by Sinan81** [comment]
The dependency 'pyosreplace' for Python < 3.3 seems to be windows specific (as the compilation requires "Windows.h") I will remove it and make sure package compiles for python@2

**Comment by adamjstewart** [comment]
Needs to be rebased now that all dependencies have been merged.

**Pull request merged**